registry/storage/manifeststore.go

```diff
@@ -39,10 +40,28 @@ func (ms *manifestStore) Get(dgst digest.Digest) (*manifest.SignedManifest, erro
 	return ms.revisionStore.get(ms.ctx, dgst)
 }
 
-func (ms *manifestStore) Put(manifest *manifest.SignedManifest) error {
+// SkipLayerVerification allows a manifest to be Put before it's
+// layers are on the filesystem
+func SkipLayerVerification(ms distribution.ManifestService) error {
+	if ms, ok := ms.(*manifestStore); ok {
+		ms.skipLayerVerification = true
```

Comment:
The way this alters the method comes across more as a side effect than an option. If the options are intended to be used this way, wouldn't a more expected pattern be...

```
SkipLayerVerification(ms) // or ms.ApplyOption(SkipLayerVerification)
ms.Put(manifest)
```